tslint/noQueryRendererImportRule.js

```diff
@@ -0,0 +1,37 @@
+// @ts-check
+
+const Lint = require("tslint")
+const ts = require("typescript")
+
+class Rule extends Lint.Rules.AbstractRule {
+  /**
+   * @param {ts.SourceFile} sourceFile
+   */
+  apply(sourceFile) {
+    // Skip storybook files.
+    if (sourceFile.fileName.includes(".story")) {
+      return []
+    }
+
+    if (
+      sourceFile.text.includes("QueryRenderer") &&
+      !sourceFile.text.includes("SystemQueryRenderer")
```

Comment:
Don't be scared of ASTs! They're really not bad. https://astexplorer.net/#/gist/9d1ac744c01ca192705f5c79c969c77f/040d8f268c48284a2aaa6f63e63b5af1f6b0f16e

Reply:
Would be happy to pair on it next week if you want.